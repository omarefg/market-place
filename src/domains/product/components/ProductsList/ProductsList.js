/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect } from "react";

export function ProductsList(props) {
  const {
    feedbackComponents,
    products,
    isLoading,
    error,
    handleAddToCart,
    productComponents,
    handleSetProducts,
  } = props;

  const { Loader, ErrorComponent } = feedbackComponents;
  const { Card, CardsContainer } = productComponents;

  useEffect(handleSetProducts, []);

  if (isLoading) {
    return <Loader />;
  } else if (error) {
    return <ErrorComponent error={error} retry={handleSetProducts} />;
  }

  return products.map(({title, content}) => (
    <CardsContainer title={title} key={title}>
      {content.map(item => (
        <Card key={item.id} onAddToCart={handleAddToCart} {...item} />
      ))}
    </CardsContainer>
  ));
}
