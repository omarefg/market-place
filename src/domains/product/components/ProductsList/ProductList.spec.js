import React from "react";
import { render } from "@testing-library/react";
import { ProductsList } from "./ProductsList";

describe("@domains/product/components/ProductList", () => {
  let defaultProps, wrapper;

  beforeEach(() => {
    defaultProps = {
      feedbackComponents: {
        Loader: jest.fn(() => null),
        ErrorComponent: jest.fn(() => null),
      },
      products: [],
      isLoading: false,
      error: "",
      handleAddToCart: jest.fn(),
      productComponents: {
        Card: jest.fn(() => null),
        CardsContainer: jest.fn(() => null),
      },
      handleSetProducts: jest.fn(),
    };

    wrapper = (props = {}) => {
      const finalProps = { ...defaultProps, ...props };
      return render(<ProductsList {...finalProps} />);
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Given a valid call it must render without crashing", () => {
    //Arrange
    const { container } = wrapper();
    // Assert
    expect(container).toMatchSnapshot();
  });
});
