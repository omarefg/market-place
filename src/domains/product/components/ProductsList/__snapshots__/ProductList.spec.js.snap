// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@domains/product/components/ProductList Given a valid call it must render without crashing 1`] = `<div />`;
